src/pybind/mgr/dashboard/services/ganesha.py

```diff
@@ -60,42 +73,36 @@ def _get_clusters_locations(cls):
                 else:
                     pool, namespace = pool_nm.split('/', 1)
 
+            if cluster in orch_result:
+                # cephadm might have set cluster settings, skip if the settings are identical.
+                orch_pool, orch_namespace = orch_result[cluster][:2]
+                if (orch_pool, orch_namespace) == (pool, namespace):
+                    logger.warning(
+                        'Duplicate cluster setting of an exsiting Orchestrator cluster is detected:'
+                        ' %s:%s/%s.', cluster, pool, namespace)
+                    continue
+                raise NFSException(
+                    'Conflicted Ganesha cluster setting: {0}:{1}/{2}. A cluster deployed by the '
+                    'Orchestrator exists: {0}:{3}/{4}.'.format(cluster, pool, namespace, orch_pool,
+                                                               orch_namespace))
```

Comment:
> would it be better allow the legacy (user-defined) config take precedence if/when the dashboard config setting has been configured?

This conflicts the previous suggestion (raising an Exception when there are duplicate cluster names).
I kept the previous suggestion.

>    rook still uses the legacy conf format, but nfs services are discover-able via the orchestrator...
    should we preserve the legacy behavior until an upgrade of each daemon has succeeded?
    etc.


Considering upgrade (migrating from legacy to Orchestrator):
- MGR is adopted first, so the Dashboard is upgraded and can support both Orchestrator and legacy Ganesha clusters.
- NFS services are adopted later. The adoption process should remove the legacy clusters definitions from the Dashboard `GANESHA_CLUSTERS_RADOS_POOL_NAMESPACE` setting.
  - There might be a window a cluster is treated as both Orchestrator and legacy cluster during the migration.
  - We should advise users that they can't use the Dashboard NFS page during the migration. Since the migration is supposed to contain daemon re-provision and can interrupt NFS I/O, this advice is reasonable.

As for Rook, the current changes in the PR doesn't work because:
- `pool`, `namespace`, and `rados_config_location` are not reported in `ServiceSpec`. (causing an exception..)
- `list_daemons` feature in Rook orchestrator seems broken.

I can add more checks to exclude Ganesha clusters reported by the Orchesetrator if these pre-requisites are not implemented. Before that, the user needs to set the `GANESHA_CLUSTERS_RADOS_POOL_NAMESPACE` setting manually to manage Ganesha clusters from Rook.